lib/entity.js

```diff
@@ -71,6 +86,21 @@ Entity.prototype.merge = function merge (snapshot) {
   return this;
 };
 
+/**
+ * Merge a property onto the instance.
+ *
+ * Given a name and a value, mergeProperty checks if the name corresponds to a
+ * function or an object.
+ * - If not either, the value is simply applied to the
+ * instance.
+ * - If value is an object and the name corresponds to a value that is
+ * also an object in the instance, it is merged.
+ * [CONFIRM] - If the name corresponds to a function, the funcition is executed with
+ * value passed as the parameter.
+ *
+ * @param  {String} name   the name of the property being merged.
+ * @param  {Object} value  the value of the property being merged.
+ */
 Entity.prototype.mergeProperty = function mergeProperty (name, value) {
   if (mergeProperties.size &&
```

Comment:
Ok, so... Let me make sure I understand.

1. `var mergeProperties = new Map();` is a global map of `types` that only gets used if one explicitly "registers" a type and its property.
2. `Entity.prototype.mergeProperty` checks wether the property being merged is registered in this global map. It does this by first checking that the instance's constructor name and the property name exist in the map (and is in fact a function) and, if so, calls the registered function on the instance.
3. If the property being merged if not in the global map, it is merged as an object or simple value.

Correct?